### PR TITLE
Replace react-infinite-scroller with react-infinite-scroll-hook

### LINKED ISSUE #2420: Replace react-infinite-scroller with react-infinite-scroll-hook
In EntitiesList.tsx, we still use `react-infinite-scroller`, which is no longer maintained.

Let's switch to https://github.com/onderonur/react-infinite-scroll-hook.

See also: #2412, #2391.

### PR DESCRIPTION
**Pull request by eemeli**
Fixes #2420

The previously skipped EntitiesList tests are now un-skipped. Minimally mocking IntersectionObserver was... interesting.

### CODE REVIEW

File: translate/src/modules/entitieslist/components/EntitiesList.test.js [hunk 1]
```diff
@@ -1,12 +1,47 @@
...
 import * as BatchActions from '~/modules/batchactions/actions';
-import { createReduxStore, mountComponentWithStore } from '~/test/store';
+
+import {
+  createDefaultUser,
+  createReduxStore,
+  mountComponentWithStore,
+} from '~/test/store';
 
 import EntitiesList, { EntitiesListBase } from './EntitiesList';
```
Comment (mathjazz) [inline translate/src/modules/entitieslist/components/EntitiesList.test.js:16]: `EntitiesListBase` is never used.

**Review by mathjazz — APPROVED** [review]
r+wc

Please remove that `EntitiesListBase` and we're good.

I also tested and didn't spot any issues.

File: translate/src/modules/entitieslist/components/EntitiesList.test.js (hunk 1 above)
Comment (eemeli) [reply translate/src/modules/entitieslist/components/EntitiesList.test.js]: Refactored the exports so EntitiesListBase no longer exists.

**Review by mathjazz — APPROVED** [review]
Great work!

**Pull request merged**